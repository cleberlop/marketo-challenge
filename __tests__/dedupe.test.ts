import Dedupe from '../src/helpers/dedupe';
import { leadsArray } from './fake-data';
import Writer from '../src/helpers/writer';

import fs from 'fs';
console.log = jest.fn();
jest.mock('fs');
let dedupe: Dedupe;
beforeEach(() => {
  jest.clearAllMocks();
  dedupe = new Dedupe();
});

describe('Dedupe class methods', () => {
  test('swap calls setListsValues', () => {
    expect.assertions(1);
    jest.spyOn(dedupe, 'setListsValues');
    dedupe.swap('id', 'email', 'prevId', 'prevEmail', leadsArray[0]);
    expect(dedupe.setListsValues).toHaveBeenCalledTimes(1);
  });

  test('setListsValues sets emailList itemsByIdList correctly', () => {
    expect.assertions(2);
    dedupe.setListsValues(leadsArray[0]._id, leadsArray[0].email, leadsArray[0]);
    expect(dedupe.emailList).toMatchSnapshot();
    expect(dedupe.itemsByIdList).toMatchSnapshot();
  });

  test('shouldSwap return correct boolean', () => {
    dedupe.deduplicate(JSON.stringify({ leads: leadsArray }));
    const shouldSwap = dedupe.shouldSwap(leadsArray[1], leadsArray[0]._id);
    expect(shouldSwap).toBe(true);
  });

  test('deduplicate successfully matches snapshot', () => {
    expect.assertions(1);
    fs.readFileSync = jest.fn().mockReturnValue('Correct File');
    dedupe.writer.writeOutputJSON = jest.fn().mockReturnValue(Promise.resolve('string'));
    return dedupe.deduplicate(JSON.stringify({ leads: leadsArray })).then(res => {
      expect(res).toMatchSnapshot();
    });
  });
});
