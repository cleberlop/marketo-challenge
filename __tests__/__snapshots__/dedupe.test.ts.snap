// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dedupe class methods deduplicate successfully matches snapshot 1`] = `"string"`;

exports[`Dedupe class methods setListsValues sets emailList itemsByIdList correctly 1`] = `
Map {
  "bog@bar.com" => "wabaj238238jdsnfsj23",
}
`;

exports[`Dedupe class methods setListsValues sets emailList itemsByIdList correctly 2`] = `
Map {
  "wabaj238238jdsnfsj23" => Object {
    "_id": "wabaj238238jdsnfsj23",
    "address": "8803 Dark St",
    "email": "bog@bar.com",
    "entryDate": "2014-05-07T17:31:20+00:00",
    "firstName": "Fran",
    "lastName": "Jones",
  },
}
`;
